backend_jobs/ingestion_pipeline/providers/image_provider_flickr.py

```diff
@@ -0,0 +1,117 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+
+from datetime import datetime
+from pipeline_lib.database_functions import get_doc_by_id
+from pipeline_lib.image_provider_interface import ImageProvider
+from pipeline_lib.additional_classes import ProviderType
+from pipeline_lib.additional_classes import VisibilityType
+from pipeline_lib.additional_classes import ImageAttributes
+import flickrapi
+
+
+class FlickrProvider(ImageProvider):
+    """
+    This class is an implementation for the ImageProvider interface.
+    """
+    def get_images(self, num_of_batches, num_of_images, query_arguments):
```

Comment:
there seem to be a disconnect with num_of_images which is sometimes hardcoded to 100, sometimes an argument?

Reply:
you are right, as of now it is hard codded